pytorch_lightning/accelerators/ddp_spawn_accelerator.py

```diff
@@ -162,7 +162,7 @@ def set_world_ranks(self, process_idx):
         self.trainer.world_size = self.trainer.num_nodes * self.trainer.num_processes
 
     def model_to_device(self, model, process_idx, is_master):
-        gpu_idx = process_idx
+        gpu_idx = self.trainer.data_parallel_device_ids[self.trainer.local_rank]
```

Comment:
does this same fix apply to the ddp_torchelastic_accelerator? are there other DDP accelerators which need this change too?

Reply:
yes, some of these accelerators like torchelastic and ddp2 should also get this fix. The problem is, I didn't want to touch these because I cannot test the fix myself, lacking multi node setup. It would be good to follow up on this.

Reply:
sorry, i actually don't think the others need this fix.

The reason is that torchelastic and co derive their process numbers from the environment, so they don't need to do this (local rank, global rank, etc).

Our DDP is a special case where we are acting like torchelastic, so we need to set this stuff up manually